src/org/scada_lts/utils/PointHierarchyUtils.java

```diff
@@ -0,0 +1,158 @@
+                                                                           DataPointDAO dataPointDAO,
+                                                                           int key)  {
+        List<PointHierarchyNode> pointNodes = getPointNodesWithAccess(user, dataPointDAO);
+        List<PointHierarchyNode> folderNodes = getFolderNodes(hierarchyDAO);
+        return getPointHierarchyNodes(key, pointNodes, folderNodes, true);
+    }
+
+    public static List<PointHierarchyNode> getPointHierarchyByKey(User user,
+                                                                  HierarchyDAO hierarchyDAO,
+                                                                  DataPointDAO dataPointDAO,
+                                                                  int key)  {
+        List<PointHierarchyNode> pointNodes = getPointNodesWithAccess(user, dataPointDAO);
+        List<PointHierarchyNode> folderNodes = getFolderNodes(hierarchyDAO);
+        return getPointHierarchyNodes(key, pointNodes, folderNodes, false);
+    }
+
+    public static PointHierarchyNode getPointHierarchyWithEmptyRoot(User user, HierarchyDAO hierarchyDAO, DataPointDAO dataPointDAO)  {
+        return getRootNode(user, hierarchyDAO, dataPointDAO);
+    }
+
+    public static PointHierarchyNode getPointHierarchyRoot(User user, HierarchyDAO hierarchyDAO, DataPointDAO dataPointDAO)  {
+        PointHierarchyNode root = getRootNode(user, hierarchyDAO, dataPointDAO);
+        cleanTree(root);
+        return root;
+    }
+
+    public static List<PointHierarchyNode> getPointNodesWithAccess(User user, DataPointDAO dataPointDAO) {
+        List<DataPointVO> dataPoints = dataPointDAO.getDataPoints();
+        List<PointHierarchyNode> pointNodes = new ArrayList<>();
+        for (DataPointVO point : dataPoints) {
+            if (Permissions.hasDataPointReadPermission(user, point)) {
+                pointNodes.add(PointHierarchyNode.pointNode(point));
+            }
+        }
+        return pointNodes;
+    }
+
+    public static void sort(List<PointHierarchyNode> nodes) {
+        for(PointHierarchyNode node: nodes) {
+            sort(node);
+        }
+    }
+
+    private static void sort(PointHierarchyNode node) {
+        if(node.isFolder())
+            node.getChildren().sort(Comparator.comparing(PointHierarchyNode::getTitle));
+    }
+
+    private static void createTree(PointHierarchyNode root,
+                                  List<PointHierarchyNode> pointAndFolderNodes) {
+        for(PointHierarchyNode node: pointAndFolderNodes) {
+            if(root.getKey() == node.getParentId()) {
+                root.getChildren().add(node);
+                if(node.isFolder()) {
+                    createTree(node, pointAndFolderNodes);
+                }
+            }
+        }
+        sort(root);
+    }
+
+    private static void cleanTree(PointHierarchyNode root) {
+        int safe = 10;
+        while (isEmpty(root) && safe > 0) {
```

Comment:
What do you think about moving that variable up as a class constant property? Or even move to move this variable to "enviroment varialbes"? ("properties.env")